packages/edge/src/vercel-edge/index.ts

```diff
@@ -126,10 +126,7 @@ function vercelMiddlewareAuth(
         return signedOutResponse(errorReason);
       }
 
-      const response = (await handler(
-        injectAuthIntoRequest(req, createSignedOutState()),
-        event,
-      )) as NextMiddlewareResult;
+      const response = await handler(injectAuthIntoRequest(req, createSignedOutState()), event);
```

Comment:
This is unrelated to the feature, but noticed that the cast was not needed. At least the linter told me so :smile: